dask_cuda/device_host_file.py

```diff
@@ -1,28 +1,59 @@
 from zict import Buffer, File, Func
 from zict.common import ZictBase
-from distributed.protocol import deserialize_bytes, serialize_bytes
+from distributed.protocol import (
+    deserialize,
+    deserialize_bytes,
+    serialize,
+    serialize_bytelist,
+    Serialized,
+)
+from dask.sizeof import sizeof
+from distributed.utils import nbytes
 from distributed.worker import weight
 
 from functools import partial
+from numba import cuda
 import os
 
 from .is_device_object import is_device_object
 
 
-def _serialize_if_device(obj):
-    """ Serialize an object if it's a device object """
-    if is_device_object(obj):
-        return serialize_bytes(obj, on_error="raise")
-    else:
-        return obj
+# Register sizeof for Numba DeviceNDArray while Dask doesn't add it
+if not hasattr(sizeof, "register_numba"):
 
+    @sizeof.register_lazy("numba")
+    def register_numba():
+        import numba
 
-def _deserialize_if_device(obj):
-    """ Deserialize an object if it's an instance of bytes """
-    if isinstance(obj, bytes):
-        return deserialize_bytes(obj)
-    else:
-        return obj
+        @sizeof.register(numba.cuda.cudadrv.devicearray.DeviceNDArray)
+        def sizeof_numba_devicearray(x):
+            return int(x.nbytes)
+
+
+# Register sizeof for Serialized while Dask doesn't add it
+if not hasattr(sizeof, "register_serialized"):
+
+    @sizeof.register(Serialized)
+    def register_serialized(obj):
+        return sum(map(nbytes, obj.frames))
```

Comment:
Can I ask you to also include a PR for this upstream in dask/distributed

Reply:
Yes, will do that for this and for the `DeviceNDArray` (into dask core) later today.